application/core/LSUserIdentity.php

```diff
@@ -67,20 +66,23 @@ public function authenticate()
             if (is_null($this->plugin)) {
                 $result->setError(self::ERROR_UNKNOWN_HANDLER);
             } else {
-                // Never allow login for non-active users.
+                // Never allow login for non-active or expired users. Check it only if user exist, plugin can create user
+                /** @var \User|null */
                 $user = User::model()->findByAttributes(array('users_name' => $this->username));
-                if ($user && (int) $user->user_status === 0) {
-                    throw new CHttpException(403, gT("You do not have permission to access this page."));
-                }
-                // Delegate actual authentication to plugin
-                $authEvent = new PluginEvent('newUserSession', $this); // TODO: rename the plugin function authenticate()
-                $authEvent->set('identity', $this);
-                App()->getPluginManager()->dispatchEvent($authEvent);
-                $pluginResult = $authEvent->get('result');
-                if ($pluginResult instanceof LSAuthResult) {
-                    $result = $pluginResult;
+                if (!is_null($user) && (!$user->isActive() || $user->isExpired())) {
```

Comment:
An idea can be move this part after plugin event newUserSession : then plugin can update user before test.

For example : 

1. For an LDAP user : check LDAP status `limesureyaccess`
2. For a Saml account : check an attribute 

It's near a new feature, but this fix both issues : didn't allow non-active or expires users + mantis issue 19319